crates/pallet-domains/src/lib.rs

```diff
@@ -2373,9 +2395,19 @@ impl<T: Config> Pallet<T> {
         opaque_bundle: &OpaqueBundleOf<T>,
         pre_dispatch: bool,
     ) -> Result<(), BundleError> {
+        let current_bundle_version =
+            T::CurrentBundleAndExecutionReceiptVersion::get().bundle_version;
+
+        // bundle version check
+        match (current_bundle_version, opaque_bundle) {
+            (BundleVersion::V0, Bundle::V0(_)) => Ok(()),
+            (BundleVersion::V1, Bundle::V1(_)) => Ok(()),
+            _ => Err(BundleError::BundleVersionMismatch),
+        }?;
```

Comment:
The bundle is carrying the ER of the parent block, if the bundle/ER version is changed from V0 to V1, that means there should be a V1 bundle carrying a V0 ER.

We need to ensure not only that such a bundle will be accepted by the runtime here, but also the client side will produce such a bundle in the first place.